docs/endpoint.md

```diff
@@ -5,14 +5,133 @@ The endpoint is one of the components provided by Tangle-accelerator, running on
+$ cd tangle-accelerator
+$ make TARGET=wp77xx legato # build endpoint as wp77xx target
+$ make TESTS=true TARGET=wp77xx legato # build endpoint as wp77xx target in test mode
+```
+
+### How to build endpoint application for native target
+
+Install required packages:
 ```
-$ bazel build //endpoint:wp7702
-$ bazel build //endpoint:sim
+$ sudo apt-get install -y   \
+    autoconf                \
+    automake                \
+    bash                    \
+    bc                      \
+    bison                   \
+    bsdiff                  \
+    build-essential         \
+    chrpath                 \
+    cmake                   \
+    cpio                    \
+    diffstat                \
+    flex                    \
+    gawk                    \
+    gcovr                   \
+    git                     \
+    gperf                   \
+    iputils-ping            \
+    libbz2-dev              \
+    libcurl4-gnutls-dev     \
+    libncurses5-dev         \
+    libncursesw5-dev        \
+    libsdl-dev              \
+    libssl-dev              \
+    libtool                 \
+    libxml2-utils           \
+    ninja-build             \
+    python                  \
+    python-git              \
+    python-jinja2           \
+    python-pkg-resources    \
+    python3                 \
+    texinfo                 \
+    unzip                   \
+    wget                    \
+    zlib1g-dev
 ```
 
+Create a workspace
+```
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Download legato by [git-repo](https://gerrit.googlesource.com/git-repo)
+```
+$ repo init -u git://github.com/legatoproject/manifest
+$ repo sync
+```
+
+You can also clone a specific version of `legato`
+```
+$ repo init -u git://github.com/legatoproject/manifest -m legato/releases/19.11.1.xml # specific legato 19.11.1 version
+$ repo sync
+```
+
+Build legato as native target
+```
+$ cd legato
+$ make localhost
+```
+
+Checkout the shell to legato shell
+```
+$ source ./framework/tools/scripts/configlegatoenv
+```
+
+Finally, go back to the root directory of tangle-accelerator. Use GNU Make to build the endpoint application.
+```
+$ make legato # build endpoint as native target
+```
+
+The endpoint will be built at `endpoint/_build_endpoint/localhost/app/endpoint/staging/read-only/bin/endpoint`
+
 ## HTTPS Connection Support
-The endpoint uses http connection as default. The message which sent to tangle-accelerator has been encrypted. So the HTTP connection would not be unsafe. To build with https connection support, add `--define https=enable` option.
+The endpoint uses HTTP connection as default. The message sent to tangle-accelerator has been encrypted, so the HTTP connection would not be unsafe. To build with HTTPS connection support, add `ENFORCE_EP_HTTPS=true` option as following.
 ```
-$ bazel build --define https=enable //endpoint:wp7702
+$ make ENFORCE_EP_HTTPS=true PEM=/path/to/pem legato
```

Comment:
What if `ENFORCE_EP_HTTPS` is set to `true` but no `PEM` available?

Reply:
Will use the default PEM file inside pem/.

Reply:
You shall mention the default treatment about PEM as well.